low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -224,16 +253,19 @@ def validate(self):
                     errno.EINVAL,
                     "%s - Argument validation failure. Global config is needed",
                     self.name)
-        global_config = self.args.config[0]
-        Conf.load('global_config', global_config)
 
-        role = Conf.get('global_config', 'release>setup')
+        # Dump global config
+        prvsnr_global_config_url = self.args.config[0]
+        self.dump_global_config(prvsnr_global_config_url)
```

Comment:
i dont get it, copy of global config can be made once, in post_install mini provisioner stage. Rest of the stages would find the copy & use it, if not found, error out with clear message. Then post-install stage should be executed again to get the copy. Remember each mini provisioner stage needs to be idempotent.

Reply:
Agree

Reply:
I agree with dumping config in post_install and referring it latest stage. But I wrote like this as other setup commands also has --config.
```         [ post_install --config [<global_config_url>] ]
            [ init --config [<global_config_url>] ]
            [ config --config [<global_config_url>] ]
            [ test --config [<global_config_url>] --plan [sanity|alerts|self_primary|self_secondary] ]
            [ reset --config [<global_config_url>] --type [hard|soft] ]
```
As per your recommendation, I should ignore --config input after post_install and refer to dumped one.
Then the command args won't make sense. Should I modify the command args too?

Reply:
i see you point @mariyappanp, as per spec this $URL is expected per mini provisioner interface, so let it be there. However it was also said later  to make a copy of this global config (using $URL) and use it. now functionally copy once is enough, which can used through rest of the mini provisioner interfaces, which we are doing and is fine.